### PR TITLE
fix ArgumentError for FastlyRails::Client#purge_by_key

### LINKED ISSUE #57: Soft Purge fails on 0.7.0
Soft purge on an ActiveRecord backed model fails and spits this out:

To reproduce - call any ActiveRecord model with:

```ruby
some_object.soft_purge
```

or 

```ruby
some_model.soft_purge_all
```

```
ArgumentError: wrong number of arguments (2 for 1)
  from /Users/my_username_redacted/.rvm/gems/ruby-2.2.2@maker/gems/fastly-rails-0.7.0/lib/fastly-rails/client.rb:11:in `purge_by_key'
  from /Users/my_username_redacted/.rvm/gems/ruby-2.2.2@maker/gems/fastly-rails-0.7.0/lib/fastly-rails.rb:26:in `purge_by_key'
  from /Users/my_username_redacted/.rvm/gems/ruby-2.2.2@maker/gems/fastly-rails-0.7.0/lib/fastly-rails/active_record/surrogate_key.rb:41:in `soft_purge'
  from (irb):7
  from /Users/my_username_redacted/.rvm/gems/ruby-2.2.2@maker/gems/railties-4.2.4/lib/rails/commands/console.rb:110:in `start'
  from /Users/my_username_redacted/.rvm/gems/ruby-2.2.2@maker/gems/railties-4.2.4/lib/rails/commands/console.rb:9:in `start'
  from /Users/my_username_redacted/.rvm/gems/ruby-2.2.2@maker/gems/railties-4.2.4/lib/rails/commands/commands_tasks.rb:68:in `console'
  from /Users/my_username_redacted/.rvm/gems/ruby-2.2.2@maker/gems/railties-4.2.4/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
  from /Users/my_username_redacted/.rvm/gems/ruby-2.2.2@maker/gems/railties-4.2.4/lib/rails/commands.rb:17:in `<top (required)>'
  from bin/rails:4:in `require'
  from bin/rails:4:in `<main>'

```

I'm on Rails 4.2.4 + Ruby 2.2.2. 

Fwiw, I was previously doing this before the soft_purge support:

```ruby
FastlyRails.client.get_service(ENV.fetch("FASTLY_SERVICE_ID")).purge_by_key(some_object.record_key, true)
```
(I originally brought it up here - https://github.com/fastly/fastly-rails/pull/56)

### PR DESCRIPTION
**Pull request by lanej**
* adjusted method signature for the `soft` argument
* add some quick tests
* Fixes #57

### CODE REVIEW

File: lib/fastly-rails/client.rb [hunk 1]
```diff
@@ -8,9 +8,8 @@ def initialize(opts={})
       super(Fastly.new(opts))
     end
 
-    def purge_by_key(key)
-      client.require_key!
```
Comment (thommahoney) [inline lib/fastly-rails/client.rb]: Why was this removed?

Comment (lanej) [inline lib/fastly-rails/client.rb]: https://github.com/fastly/fastly-ruby/blob/master/lib/fastly/service.rb#L73

Comment (thommahoney) [inline lib/fastly-rails/client.rb]: 👍

**Comment by samratjp** [comment]
Cool beans! When do you guys think this PR will be merged?

**Comment by gschorkopf** [comment]
LGTM

**Pull request merged**